storage/gcp/gcp_test.go

```diff
@@ -304,9 +305,72 @@ func TestBundleRoundtrip(t *testing.T) {
 	}
 }
 
+func TestPublishCheckpoint(t *testing.T) {
+	ctx := context.Background()
+
+	close := newSpannerDB(t)
+	defer close()
+
+	s, err := newSpannerSequencer(ctx, "projects/p/instances/i/databases/d", 1000)
+	if err != nil {
+		t.Fatalf("newSpannerSequencer: %v", err)
+	}
+
+	for _, test := range []struct {
+		name            string
+		cpModifiedAt    time.Time
+		publishInterval time.Duration
+		wantUpdate      bool
+	}{
+		{
+			name:            "works ok",
+			cpModifiedAt:    time.Now().Add(-15 * time.Second),
+			publishInterval: 10 * time.Second,
+			wantUpdate:      true,
+		}, {
+			name:            "too soon, skip update",
+			cpModifiedAt:    time.Now().Add(-5 * time.Second),
+			publishInterval: 10 * time.Second,
+			wantUpdate:      false,
+		},
```

Comment:
Add a test for the case where no checkpoint exists yet?

Reply:
That's already implicitly covered - `init()` creates & publishes a zero-sized checkpoint if one doesn't already exist.